examples/neb_analysis.lua

```diff
@@ -0,0 +1,209 @@
+      IOprint("\nLUA NEB Analysis for image :".. image_label .. "calculator")  
+      -- This is only reached one time, and that it as the beginning...
+      -- be sure to set the corresponding values
+      siesta.geom.xa = images[1].R* Unit.Ang  --NEB.initial.R * Unit.Ang
+      siesta.send({"geom.xa"})
+      IOprint("\nLUA NEB Analysis for image Coordinates :\n")
+      IOprint(images[1].R* Unit.Ang )
+
+  end
+
+  if siesta.state == siesta.MOVE then
+      -- Here we are doing the actual LBFGS algorithm.
+      -- We retrieve the current coordinates, the forces
+      -- and whether the geometry has relaxed
+      siesta.receive({"MD.Relaxed"})
+      IOprint("\nLUA/NEB MOVE state STARTED\n")
+      siesta.MD.Relaxed = true
+      siesta.send({"MD.Relaxed"})
+  end
+  
+    if siesta.state == siesta.ANALYSIS_AFTER then
+      IOprint("\nLUA/NEB AFTER ANALYSIS state STARTED\n")
+      siesta_update_Analysis(image_number)
+     --os.execute("cp " .. label .. ".PDOS" .. " " .. label .. ".PDOS.3")
+   --   IOprint("\nLUA/NEB MOVE state FINISHED\n")
+
+      siesta_cleanup(cleanup)
+      --ret_tbl = siesta_move(siesta)
+  end
+   
+end
+ 
+function siesta_update_DM(image_number)
+
+   if not siesta.IONode then
+      -- only allow the IOnode to perform stuff...
+      return
+   end
+   -- Move about files so that we re-use old DM files
+   local DM = label .. ".DM"
+   local current_DM = DM .. "." .. tostring(image_number)  
+   -- Saving initial DM
+   IOprint("Saving " .. current_DM .. " to " .. DM)
+   os.execute("cp " .. current_DM .. " " .. DM)   
+   
+end
+
+function siesta_update_Analysis(image_number)
+
+   if not siesta.IONode then
+      -- only allow the IOnode to perform stuff...
+      return
+   end
+   -- Move about files so that we re-use old DM files
+   local PDOS = label .. ".PDOS"
+   local DOS = label .. ".DOS"
+   local EIG = label .. ".EIG"
+   local BAND = label .. ".bands"
+   local FA = label .. ".FA"
+   local FAC = label .. ".FAC"
+   local PDOSKP = label .. ".PDOS.KP"
+   local PDOSxml = label .. ".PDOS.xml"
+   local XV = label .. ".XV"
+   local XYZ = label .. ".xyz"
+   local FORCE_STRESS = "FORCE_STRESS"
+   local current_PDOS = PDOS .. "." .. tostring(image_number)
+   local current_DOS = DOS .. "." .. tostring(image_number)
+   local current_EIG = EIG .. "." .. tostring(image_number) 
+   local current_BAND = BAND .. "." .. tostring(image_number) 
+   local current_FA = FA .. "." .. tostring(image_number) 
+   local current_FAC = FAC .. "." .. tostring(image_number) 
+   local current_PDOSKP = PDOSKP .. "." .. tostring(image_number) 
+   local current_PDOSxml = PDOSxml .. "." .. tostring(image_number) 
+   local current_FORCE_STRESS = FORCE_STRESS .. "." .. tostring(image_number) 
+   local current_XV = XV .. "." .. tostring(image_number) 
+   local current_XYZ = XYZ .. "." .. tostring(image_number) 
+   -- Saving initial PDOS
+   IOprint("Saving " .. PDOS .. " to " .. current_PDOS)
```

Comment:
There is no need to saving these. The PDOS will only be calculated for the final step, so lots of these seems redundant?

Reply:
Also, I don't think these particularly fits NEB analysis, rather a generic "storage" class could be used which may be useful in other situations as well.